cosmos-admin/README.md

```diff
@@ -35,15 +46,168 @@ If for any reason the data copying is interrupted (for instance, the communicati
+
+`get_user_stats.sh` is parameterized by:
+
+* Host running the MySQL server
+* Port where the MySQL server listens for requests. Typically, TCP/3306 port.
+* Database name. Usually, `cosmos`.
+* MySQL user allowd to insert data within the `cosmos` database, `cosmos_user` table.
+* Pasword for the MySQL user.
+* ssh log file (usually, `/var/log/secure`).
+
+In a shell:
+
+    $ ./get_user_stats.sh <mysql_host> <mysql_port> <db_name> <mysql_user> <mysql_password> <ssh_logs>
+
+Since the above mentioned statistics change along the time, it is useful to schedule this script execution in a periodic fashion, let's say at least once per day. You can use the `crontab` for achieving this.
+
+At the same time, the system log rotation must be done at the same frequency this script is run. The reason is the ssh log file is used in order to obtain the number of successful and failed connections; therefore, if this is done in a per day fashion, then the system logs must be rotated once per day; if the statistics are got once per week, then the system logs must be rotated once per week as well; and so on. If you don't want to rotate all the system logs at the same frequency this script is executed, at least you must syncronize the ssh log file among all the system log files.
+
+[Top](#top)
+
+##<a name="osprogramming"></a>OS programming regarding the administrative scripts
+
+###<a name="crontab"></a>Crontab
+`crontab` is the tool used by any Unix-based system to schedule task executions. In this case, we will use this application in order to schedule some of the Cosmos administrative scripts in a periodic fashion.
+
+Scheduling is done through a file that can be edited by typing:
+
+    $ crontab -e
+    
+Then, and according to the `crontab` specification, add the following lines:
+
+```
+59 23 * * * /usr/local/cron_scripts/get_user_stats.sh
+00 00 * * * run-parts /etc/cron.daily
+00 00 * * 5 run-parts /etc/cron.weekly
+00 00 1 * * run-parts /etc/cron.monthly
+```
+
+The first line is about executing the `get_user_stats.sh` script each day at 23:59. The other lines must be added if not existing, and are related to the execution of certain daily, weekly and monthly tasks specified under `/etc/cron.daily`, `/etc/cron.weekly` and `/etc/cron.month` respectively; this includes the rotation of log files (how to configure the log rotation for Cosmos is explained in the next section).
+
+Please observe the `get_user_stats.sh` script is executed exactly one minute before the daily rotation is performed. This is because the `get_user_stats.sh` script must take a snapshot of the `/var/log/secure` file, which must be rotating in a daily fashion (see next section for more details).
+
+[Top](#top)
+
+###<a name="logrotation"></a>Log rotation
+Log rotation is usually performed by the `logrotate` application. Since this should be installed and already running by default in all Unix-based machines, most probably you will have the following elements in your file system:
```

Comment:
I will include the commant to check this... just in case some system doesn't have it installed (I'm not sure, but I think that in some case I have seen something like that).